ui-v2/src/hooks/global-concurrency-limits.ts

```diff
@@ -77,4 +82,148 @@ export const useGetGlobalConcurrencyLimit = (id_or_name: string) =>
 // ----- ✍🏼 Mutations 🗄️
 // ----------------------------
 
-// TODO:
+/**
+ * Hook for deleting a global concurrency limit
+ *
+ * @returns Mutation object for deleting a global concurrency limit with loading/error states and trigger function
+ *
+ * @example
+ * ```ts
+ * const { deleteGlobalConcurrencyLimit } = useDeleteGlobalConcurrencyLimit();
+ *
+ * // Delete a  global concurrency limit by id or name
+ * deleteGlobalConcurrencyLimit('id-to-delete', {
+ *   onSuccess: () => {
+ *     // Handle successful deletion
+ *   },
+ *   onError: (error) => {
+ *     console.error('Failed to delete global concurrency limit:', error);
+ *   }
+ * });
+ * ```
+ */
+export const useDeleteGlobalConcurrencyLimit = () => {
+	const queryClient = useQueryClient();
+	const { mutate: deleteGlobalConcurrencyLimit, ...rest } = useMutation({
+		mutationFn: (id_or_name: string) =>
+			getQueryService().DELETE("/v2/concurrency_limits/{id_or_name}", {
+				params: { path: { id_or_name } },
+			}),
+		onSuccess: () => {
+			// After a successful deletion, invalidate the listing queries only to refetch
+			return queryClient.invalidateQueries({
+				queryKey: queryKeyFactory.lists(),
```

Comment:
Do we want to invalidate the details query for the deleted limit like the update hook does? Or delete the detail entry from the cache if it exists?

Reply:
I don't think its necessary, but we can place that invalidation later in the future.
Seeing how the current ui does a fetch automatically per each item on the list, I'm afraid of fetching all limit details.

eg: if `/filter` brings 20 runs, I'm trying to avoid having 20 individual API calls upon deleting just one

Reply:
Also, comparing the /filter APIs and the payoad for `GET`, the data looks the same?

Maybe I can just remove the `GET` API from the query key factory, and just invalidate all